### PR TITLE
@aliases Endpoint

### PR DESCRIPTION
**Pull request by Petchesi-Iulian**
For #1393

### CODE REVIEW

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by avoinea** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/services/aliases/get.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        batch = RedirectsControlPanel(self.context, self.request).redirects()
+        redirects = [entry for entry in batch]
+
+        for redirect in redirects:
+            del redirect["redirect"]
+            redirect["datetime"] = datetimelike_to_iso(redirect["datetime"])
+        self.request.response.setStatus(201)
+
+        self.request.form["b_start"] = "0"
+        self.request.form["b_size"] = "1000000"
```
Comment (tisto) [inline src/plone/restapi/services/aliases/get.py:46]: I am wondering if we shouldn't go the extra mile of implementing batching here. I can imagine that the number of redirects can become quite significant even on mid-size websites.

File: docs/source/aliases.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+---
+html_meta:
+  "description": "Aliases - a mechanism to redirect old URLs to new ones."
```
Comment (tisto) [inline docs/source/aliases.md:3]: Any particular reason why we call this "aliases" instead of "redirects"?

Comment (avoinea) [reply docs/source/aliases.md:3]: @tisto I think it started from Plone Classic, where the view on context is called `@@manage-aliases`

I don't have a strong opinion on it, but I think I like more `aliases` than `redirects`.

File: src/plone/restapi/services/aliases/get.py (hunk 1 above)
Comment (avoinea) [reply src/plone/restapi/services/aliases/get.py:46]: @tisto Do you suggest to set the default `b_size` to `50`?

**Comment by fredvd** [comment]
@jenkins-plone-org please run jobs

File: docs/source/aliases.md (hunk 2 above)
Comment (tisto) [reply docs/source/aliases.md:3]: I don't have a strong opinion here either. Maybe others want to weigh in here. We should just be consistent in Plone Classic and Volto.

**Comment by avoinea** [comment]
@fredvd Are the failing jenkins test still not-related to this PR or it's a real issue?

**Comment by avoinea** [comment]
@jenkins-plone-org please run jobs

**Comment by fredvd** [comment]
> Are the failing jenkins test still not-related to this PR or it's a real issue?

@avoinea These are not related to this PR, there are still a few flakey robot tests left on plone.schema editor and the generic Fields robot tests that fail occasionally. Those are also failing in other tests. 

I was fumbling around a bit to check the failures/success for this specific PR on the Pull request jobs on jenkins, just learned you can paste the PR URL in the filter on the PR job page to see them:

pr filter on url

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by Petchesi-Iulian** [comment]
@jenkins-plone-org please run jobs

**Comment by avoinea** [comment]
@jenkins-plone-org please run jobs

**Review by avoinea — APPROVED** [review]

**Comment by tisto** [comment]
@Petchesi-Iulian @avoinea just wondering. The current @aliases endpoint does not allow filtering by aliases that have been added manually or automatically, is that correct?

**Comment by avoinea** [comment]
@tisto Yes, I think you're right. There is no way to filter them by the `manual` criteria :thinking:

**Comment by tisto** [comment]
@avoinea thanks for the feedback! I created an issue for this:

https://github.com/plone/plone.restapi/issues/1791

I also noticed that we do not have any way to filter for a date range. I also created a ticket for this:

https://github.com/plone/plone.restapi/issues/1792

We should also add the functionality to upload/download aliases as CSV file:

https://github.com/plone/plone.restapi/issues/1793

Once we have the REST API endpoints in place, we could enhance the Volto URL control panel to expose this new functionality.

Are you folks interested in those functionalities? We (at kitconcept) are currently working with the ClassicUI controlpanel but this is not a long-term solution. Therefore I'd love to see if we would catch up with Volto.

**Pull request merged**